core/storage/suggestion/gae_models.py

```diff
@@ -479,17 +480,29 @@ def get_in_review_translation_suggestions(
                 As a user cannot review their own suggestions, suggestions
                 authored by the user will be excluded.
             language_codes: list(str). The list of language codes.
+            exp_ids: list(str). Exploration IDs matching the target ID of the
+                translation suggestions.
 
         Returns:
             list(SuggestionModel). A list of suggestions that are of the given
             type, which are in review, but not created by the given user.
         """
-        return cls.get_all().filter(datastore_services.all_of(
-            cls.status == STATUS_IN_REVIEW,
-            cls.suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
-            cls.author_id != user_id,
-            cls.language_code.IN(language_codes)
-        )).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
+
+        if len(exp_ids) > 0:
+            return cls.get_all().filter(datastore_services.all_of(
+                cls.status == STATUS_IN_REVIEW,
+                cls.suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                cls.author_id != user_id,
+                cls.language_code.IN(language_codes),
+                cls.target_id.IN(exp_ids)
+            )).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
+        else:
```

Comment:
We need to handle an edge case where we switch to an empty topic which has zero explorations in it . Currently we display all suggestion for empty topics. When we should be displaying no suggestions.

Screenshot below explains it better. 

![Screen Shot 2022-03-03 at 3 36 33 PM](https://user-images.githubusercontent.com/35385366/156672635-d7f28ae6-987a-4fe3-b3af-a05af8a2c9d8.png)

Reply:
Handled this edge case now.

Reply:
Just confirming though, I believe this would only occur if we pass EMPTY_TOPIC? 

Because, IIUC, ideally, we should not have topics in backend datastore if there are no explorations associated with the topic?

Reply:
That's not correct -- the situation you describe can happen in the case of topics which are still "in draft". Skills, etc. may have been added to them already, but the explorations may not have been added yet.

Reply:
Oh, I see. This makes sense.